lib/datadog/profiling/exporter.rb

```diff
@@ -52,7 +52,12 @@ def initialize(
 
       def flush
         worker_stats = @worker.stats_and_reset_not_thread_safe
-        start, finish, compressed_pprof = pprof_recorder.serialize
+        serialization_result = pprof_recorder.serialize
+        if serialization_result.nil?
+          Datadog.logger.debug('Skipped exporting profiling events due to failed serialization')
+          return
+        end
+        start, finish, compressed_pprof, profile_stats = serialization_result
         @last_flush_finish_at = finish
 
         return if compressed_pprof.nil? # We don't want to report empty profiles
```

Comment:
I think these two checks are redundant?

Reply:
Yeah they would only be non-redundant if `compressed_pprof`  could be nil with everything else not nil but it comes from https://github.com/DataDog/dd-trace-rb/blob/0221fbebc9849f7a2b30984b1670eea1eeaf8ad6/ext/datadog_profiling_native_extension/stack_recorder.c?plain=1#L560 which is guaranteed not to be `Qnil`